### ISSUE TITLE
Block upgrade before changes when the upgrade is forbidden

### ISSUE THREAD

**Issue by vdanjean** [nextcloud/docker#1809]
Hi,

I'm not sure if this is a nextcloud docker image bug or a nextcloud server one.
I'm using official nextcloud docker images through a docker-compose file that was referencing the 'nextcloud' docker image.

Due to various reasons, I was running nextcloud 22, and I missed nextcloud 23.
Today, I ran 
```
docker-compose build --pull
docker-compose up -d
```
And it have been a really big mess: skipping a major upgrade is not allowed by nexcloud, but trying to change the image back to the nextcloud 23 (using 'nextcloud:23-apache') did not work either, because some files have been modified and were referencing the 24 version ('no downgrade allowed' whereas the 22->24 migration has not been done).

With the help of internet, I found:
- https://help.nextcloud.com/t/cant-start-nextcloud-because-the-version-of-the-data-is-higher-than-the-docker-image-version-and-downgrading-is-not-supported/109438 that gives me the trick about `version.php`
- https://github.com/nextcloud/server/issues/8240 that give me the trick about manually replacing the `apps/` folder (with the 23 image, the apps were still the one from the 24 version, as seen in `apps/files/appinfo/info.xml` with min/max version of nextcloud).

So, I'm not sure if the problem comes from the update of the docker image with docker compose, or if the problem comes from nextcloud server. But it would be very appreciated if the version check were done before any changes are done. Or if getting the 23 image really install the 23 version (it was not the case for `version.php` and the `apps/*` at least, I'm not a docker specialist so I do not know why).

That said, I removed the reference to the `nextcloud:apache` and I'm using `nextcloud:MAJOR-apache` now, to avoid to suffer this upgrade problem again. Major version upgrade will need that I manually enable it in my `docker-compose.yml` file. I think that it could be useful to add this (avoid images without major version) in the documentation if early blocking the upgrade is not possible/easy.

Regards
  Vincent

**Comment by wetneb** [comment]
I agree that this is quite frustrating UX - I ended up in the state described above, not being able to run NextCloud from any image, and not knowing what to change in the volumes for any version to accept tot boot. Quite stressful experience.

So I have proposed a change to check for this before running any step of the update path: #2018.
